goinverse-oss/app
# Persist playback state

## Description
- Save a portion of the playback state
  - Restores currently-playing item on rehydrate
- Save the playback state per-item for podcast episodes
  - Restores the playback state for each episode individually, even if it wasn't the most recently playing
- New unit tests for the playback epic
  - Super rough. I'm finding it quite tricky to unit test RxJS code, but I'm starting to get my head around it. Should revisit in the future.

## Motivation and Context
Closes #132.

## How Has This Been Tested?
- New unit tests pass
- Verified that playback status is saved for podcast episodes
  - Loaded a podcast episode
  - Seeked to the middle
  - Loaded another episode
  - Seeked to the middle
  - Switch back to the previous episode
  - Observe saved playback position
  - Repeat
- Verified that playback status is restored on rehydrate
  - Same steps as above, but starting by reloading the app
